### ISSUE TITLE
[Server Request]: Ark: Survival Ascended

### ISSUE THREAD

**Issue by naoestu** [GameServerManagers/LinuxGSM#4350]
### Game server

Ark: Survival Ascended

### Linux support

- [X] Yes

### Steam

Yes

### Steam appid

Client: 2399830
Server: 2157740 or 2430930

### Guides

Ark Survival Ascended is coming this week. 
It will be perfect if you can add it to the list.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by dgibbs64** [comment]
Cant see an issue as long as the appid is provided. I cant see how to even buy the game currently

**Comment by naoestu** [comment]
yeah, i already prepared it on my side, i'm just waiting for the release so i can add the appid, 
test if it is ok and create a pull request (if needed).

**Comment by naoestu** [comment]
Steam Store: https://store.steampowered.com/app/2399830/ARK_Survival_Ascended/
Dedicated server appid will be 2157740

**Comment by BrandtNick** [comment]
> > yeah, i already prepared it on my side, i'm just waiting for the release so i can add the appid, test if it is ok and create a pull request (if needed).
> 
> 2430930

That appId is Windows only according to https://steamdb.info/app/2430930/info/

**Comment by dgibbs64** [comment]
Also I want to confirm do we need a copy of the game client to run get the dedi server via steamcmd?

**Comment by Claiyc** [comment]
> Also I want to confirm do we need a copy of the game client to run get the dedi server via steamcmd?

Yes, definitely. And that will also not change in the near future.

**Comment by dgibbs64** [comment]
I am going to need to get a license sorted. I have 2 steam accounts, my personal one and the other for use with steamcmd. So I might need to buy the game twice 😢

**Comment by valentin-gosselin** [comment]
Oh really? Do we need to have two game licenses for this? Can't our first license work simultaneously for the server and for us playing?

**Comment by Claiyc** [comment]
> I am going to need to get a license sorted. I have 2 steam accounts, my personal one and the other for use with steamcmd. So I might need to buy the game twice 😢

Nitrado says, people with large communities can apply for a special license (I think you posted a screenshot of that yourself on dc). Don’t know if linuxgsm may be eligible for that…

**Comment by naoestu** [comment]
**Update**:  Now we can host how many servers as we want (not for commercial use) ;)
(no steam account needed)

**Comment by dgibbs64** [comment]
Thats great news. I now need to start parameters to get going

**Comment by naoestu** [comment]
> Thats great news. I now need to start parameters to get going

Basically everything is like ASE, you only need to change the map name to "TheIsland_WP" and the server binary name ;)

**Comment by naoestu** [comment]
ASA is UE5, not UE4 :)

**Comment by naoestu** [comment]
Also "ShooterGameServer" is now "ArkAscendedServer".

**Comment by Toaaa** [comment]
So can you run an ARK: Survival Ascended dedicated server or not? Because according to SteamDB [it is only available for Windows](https://steamdb.info/app/2430930/) 🤔

**Comment by dgibbs64** [comment]
Testing is still ongoing

**Comment by dgibbs64** [comment]
Sadly io have checked the server files and there are no Linux binarys yet. For now ASA is Windows only. I am assuming that this will change but for now we are waiting on the ARK developers to release them.

**Comment by Ch4r0ne** [comment]
Hi all, I need some information about what exactly you want, I have started to build a basic server manager for Windows, but if more interested in Linux, what features should it have web GUI?...

https://github.com/Ch4r0ne/ARK-Ascended-Server-Manager

**Comment by dgibbs64** [comment]
It's worth looking at WindowsGSM project

**Comment by cdp1337** [comment]
I can confirm that despite being a Windows-only server, it DOES work with proton!  (At least on Debian 12 and GE's Proton...)

ie: https://github.com/cdp1337/ARKSurvivalAscended-Linux/blob/main/server-install-debian12.sh

My install doesn't include any of the nicities of LGSM, but should serve as a basis for building an installer for your project.

**Comment by Gragster** [comment]
> I can confirm that despite being a Windows-only server, it DOES work with proton! (At least on Debian 12 and GE's Proton...)
> 
> ie: https://github.com/cdp1337/ARKSurvivalAscended-Linux/blob/main/server-install-debian12.sh
> 
> My install doesn't include any of the nicities of LGSM, but should serve as a basis for building an installer for your project.

I got it working on debian bookworm, you are the MAN! also love to LinuxGSM <3

**Comment by hydrixion** [comment]
> I can confirm that despite being a Windows-only server, it DOES work with proton! (At least on Debian 12 and GE's Proton...)
> 
> ie: https://github.com/cdp1337/ARKSurvivalAscended-Linux/blob/main/server-install-debian12.sh
> 
> My install doesn't include any of the nicities of LGSM, but should serve as a basis for building an installer for your project.

Could you tell me how to find the port this in running on? it start as it should on my server but it seems that it doesnt use the standart ports

**Comment by cdp1337** [comment]
> Could you tell me how to find the port this in running on? it start as it should on my server but it seems that it doesnt use the standart ports

The game server runs on UDP/7777 and RCON by default runs on TCP/27015.  In my testing I've had difficulty getting the game port to actually change, don't know if it's hard coded or I just haven't found the right ini option...

If you've done `sudo apt install net-tools`, you should have netstat available to check ports.

`sudo netstat -natvp` -- show TCP ports and the processes listening.

`sudo netstat -nauvp` -- show UDP ports and the processes listening.

**Comment by hydrixion** [comment]
> > Could you tell me how to find the port this in running on? it start as it should on my server but it seems that it doesnt use the standart ports
> 
> The game server runs on UDP/7777 and RCON by default runs on TCP/27015. In my testing I've had difficulty getting the game port to actually change, don't know if it's hard coded or I just haven't found the right ini option...
> 
> If you've done `sudo apt install net-tools`, you should have netstat available to check ports.
> 
> `sudo netstat -natvp` -- show TCP ports and the processes listening.
> 
> `sudo netstat -nauvp` -- show UDP ports and the processes listening.

thx for the answer right after asking i found the ports and opened them funny enough that i cant find the server in steam favorites but in unofficial server section in ASA my server works now thx for your work man :)

**Comment by sdinitto** [comment]
> I can confirm that despite being a Windows-only server, it DOES work with proton! (At least on Debian 12 and GE's Proton...)
> 
> ie: https://github.com/cdp1337/ARKSurvivalAscended-Linux/blob/main/server-install-debian12.sh
> 
> My install doesn't include any of the nicities of LGSM, but should serve as a basis for building an installer for your project.

I can also confirm using your method it works. I didn't run your script but dissected it to customize my own runtime on another distro and it launched without any problems.

**Comment by hydrixion** [comment]
Btw got it that running on debian 11 so its not only for debian 12 ^^

**Comment by mrk3767** [comment]
I'm taking that script and attempting to dockerize it in Alpine linux... Fun times over here. I know this isn't really relevant to the purpose of this thread, but wanted to drop some info from my journey.

Everything seems to go well until I actually execute proton. I'm doing everything as root (rather than creating a steam user). I'm getting this lovely file not found error:
`cca4535bc140:~/Steam/steamapps/common/ARK Survival Ascended Dedicated Server/ShooterGame/Binaries/Win64# /bin/bash -E -c "$STEAMDIR/compatibilitytools.d/$PROTON_NAME/proton run ArkAscendedServer.exe TheIsland_WP?listen"
Traceback (most recent call last):
  File "/root/Steam/compatibilitytools.d/GE-Proton8-22/proton", line 1913, in <module>
    rc = g_session.run()
         ^^^^^^^^^^^^^^^
  File "/root/Steam/compatibilitytools.d/GE-Proton8-22/proton", line 1871, in run
    rc = self.run_proc(adverb + argv + sys.argv[2:] + self.cmdlineappend)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/Steam/compatibilitytools.d/GE-Proton8-22/proton", line 1842, in run_proc
    return subprocess.call(args, env=local_env, stderr=self.log_file, stdout=self.log_file)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/subprocess.py", line 389, in call
    with Popen(*popenargs, **kwargs) as p:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.11/subprocess.py", line 1950, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/root/Steam/compatibilitytools.d/GE-Proton8-22/files/bin/wine64'`

I can confirm that this file does exist and has contents:

`cca4535bc140:~/Steam/steamapps/common/ARK Survival Ascended Dedicated Server/ShooterGame/Binaries/Win64# ls -l /root/Steam/compatibilitytools.d/GE-Proton8-22/files/bin
total 796
-rwxr-xr-x    1 root     root         35864 Oct 30 19:27 cabextract
-rwxr-xr-x    1 root     root          2133 Oct 30 19:26 msidb
-rwxr-xr-x    1 root     root         15560 Oct 30 19:26 wine
-rwxr-xr-x    1 root     root         17664 Oct 30 19:26 wine-preloader
-rwxr-xr-x    1 root     root         16712 Oct 30 19:27 wine64
-rwxr-xr-x    1 root     root         18488 Oct 30 19:27 wine64-preloader
-rwxr-xr-x    1 root     root        695208 Oct 30 19:27 wineserver`

I can also verify that python3 can actually see this file at that path as well:

`cca4535bc140:~/Steam/steamapps/common/ARK Survival Ascended Dedicated Server/ShooterGame/Binaries/Win64# python3
Python 3.11.6 (main, Oct  4 2023, 06:22:18) [GCC 12.2.1 20220924] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.listdir("/root/Steam/compatibilitytools.d/GE-Proton8-22/files/bin")
['wineserver', 'wine-preloader', 'cabextract', 'wine64-preloader', 'msidb', 'wine64', 'wine']`

This is leading me to believe that the virtual environment proton executes in doesn't have the path I've got it installed to mounted. I found a few references stating you could use env vars to specify extra directories to be mounted, but that didn't pan out either.

If everyone is getting it working on Debian, I guess I can try it again going down that path haha

**Comment by poley5** [comment]
> I can confirm that despite being a Windows-only server, it DOES work with proton! (At least on Debian 12 and GE's Proton...)
> 
> ie: https://github.com/cdp1337/ARKSurvivalAscended-Linux/blob/main/server-install-debian12.sh
> 
> My install doesn't include any of the nicities of LGSM, but should serve as a basis for building an installer for your project.

I got it to launch the server but when I try to join the server it says "Joining failed Connection timeout" There were a couple errors when launching it but if there was a problem the server shouldnt have started up right?

**Comment by Lanwoah** [comment]
I can't get it to showup in steam servers, do you guys set ports or only use default ones?

**Comment by azixus** [comment]
> I can't get it to showup in steam servers, do you guys set ports or only use default ones?
You only need a single port for the server to show up. It's port 7777/udp by default.

@mrk3767 I have a working container if you want to reuse some parts (see https://github.com/azixus/ARK_Ascended_Docker)

**Comment by Lanwoah** [comment]
Thats very nice of you and i'll keep it in consideration but i'd like to keep using what i use now  for now, but it seems like my server is not running properly, its not updating the world file

**Comment by mrk3767** [comment]
> > I can't get it to showup in steam servers, do you guys set ports or only use default ones?
> > You only need a single port for the server to show up. It's port 7777/udp by default.
> 
> @mrk3767 I have a working container if you want to reuse some parts (see https://github.com/azixus/ARK_Ascended_Docker)

😍 This is exactly what I was going for! I was trying with Alpine and running down a rabbit hole. Thank you!

**Comment by hydrixion** [comment]
> I can't get it to showup in steam servers, do you guys set ports or only use default ones?

I cant see my server in steam servers too your server will use default ports 7777 but u cant find in in steam
Just name your server in the GuS.ini and then search the name in unofficial server thats what me helped atleast getting in running and steam still doesnt show me my server

**Comment by hydrixion** [comment]
> > I can confirm that despite being a Windows-only server, it DOES work with proton! (At least on Debian 12 and GE's Proton...)
> > ie: https://github.com/cdp1337/ARKSurvivalAscended-Linux/blob/main/server-install-debian12.sh
> > My install doesn't include any of the nicities of LGSM, but should serve as a basis for building an installer for your project.
> 
> I got it to launch the server but when I try to join the server it says "Joining failed Connection timeout" There were a couple errors when launching it but if there was a problem the server shouldnt have started up right?

how many times u tried joining your server? when i download mods from my server or have a bit to much and it takes him a bit time to load them i get this error sometimes too sometimes even 2-3 times in a row but then it works i hope that will help you

**Comment by Lanwoah** [comment]
Yeah my server was working apparently i just had to go ingame with unofficial, i could join so it works

**Comment by poley5** [comment]
> > > I can confirm that despite being a Windows-only server, it DOES work with proton! (At least on Debian 12 and GE's Proton...)
> > > ie: https://github.com/cdp1337/ARKSurvivalAscended-Linux/blob/main/server-install-debian12.sh
> > > My install doesn't include any of the nicities of LGSM, but should serve as a basis for building an installer for your project.
> > 
> > 
> > I got it to launch the server but when I try to join the server it says "Joining failed Connection timeout" There were a couple errors when launching it but if there was a problem the server shouldnt have started up right?
> 
> how many times u tried joining your server? when i download mods from my server or have a bit to much and it takes him a bit time to load them i get this error sometimes too sometimes even 2-3 times in a row but then it works i hope that will help you

Probably tried about 10-20 times over the last few days

**Comment by askrabal** [comment]
I've been using this script to run a server for the past few weeks and it's been working pretty well. I used LGSM for ARK SE and modeled some of the layout after that implementation. I figured it might be helpful getting LGSM support for ASA.

```shell
#!/bin/bash

mkdir -p ~/.steam/steam
export STEAM_LIBRARY="$(realpath ~/.steam/steam/)"

app_id=2430930
proton_id=2348590
server_files="$(realpath ./serverfiles)"
proton_dir="$(realpath ./serverfiles/Proton/)"

# must login with account for proton
if [[ "$1" == "u" ]];then
steamcmd \
  +force_install_dir ${proton_dir} \
  +login ${STEAM_USER} \
  +app_update ${proton_id} \
  validate \
  +quit
fi

# this works
if [[ "$1" == "u" ]];then
steamcmd \
  +@sSteamCmdForcePlatformType windows \
  +force_install_dir ${server_files} \
  +login anonymous \
  +app_update ${app_id} \
  validate \
  +quit
fi

export STEAM_COMPAT_DATA_PATH="${server_files}/compatdata/${app_id}"
export STEAM_COMPAT_CLIENT_INSTALL_PATH="${STEAM_COMPAT_DATA_PATH}"
mkdir -p "${STEAM_COMPAT_DATA_PATH}"
echo "Installed and validated"
echo "Config files at: ${server_files}/ShootGame/Saved/Config/WindowsServer/"

stop_server ()
{
  pgrep -f ^Z.*ArkAscendedServer.exe | xargs kill -SIGINT
}

trap "stop_server" SIGTERM SIGINT

# proton verbs waitforexitandrun, runinprefix
"${proton_dir}"/proton runinprefix ${server_files}/ShooterGame/Binaries/Win64/ArkAscendedServer.exe TheIsland_WP?listen?RCONPort=27020?RCONEnabled=True?MaxPlayers=10?SessionName="${SESSION_NAME}" -WinLiveMaxPlayers=10 -nosteamclient -NoBattlEye 2> /dev/null &
wait $!
echo -e "\nServer shutdown"

```

**Comment by askrabal** [comment]
@dgibbs64 Sorry for the bad pull request, I meant to make it in my own mirror. 

I understand the desire for linux binaries, but I was wondering if launching it with wine/proton would work within the LGSM framework? I've had great success running the windows binary with proton, but it would be great to have all the niceties of LGSM.

**Comment by mrk3767** [comment]
This is very likely due to how host files/folders get mounted into
wine/proton. It's a very finicky installation to get working right... I've
been using this image which has got proton installed and functioning
properly: https://github.com/azixus/ARK_Ascended_Docker

~Meik

Click here to schedule a meeting <https://kenworthy.io/schedule>

**Comment by russh3l** [comment]
@mrk3767 yes but I want to do it without using docker. is not it possible?

**Comment by lucassr-dev** [comment]
Could anyone do this for ARK SE? for him to install the ark server windows and run it on proton?

**Comment by dgibbs64** [comment]
No native Linux support still. I am going to close this. Please open a new issue when linux is supported
